plugins/modules/redfish_firmware_rollback.py

```diff
@@ -0,0 +1,340 @@
+    if module.params["reboot"] and preview_uri:
+        payload = {"ResetType": "ForceRestart"}
+        job_resp_status, reset_status, reset_fail = wait_for_redfish_reboot_job(redfish_obj, SYSTEM_RESOURCE_ID,
+                                                                                payload=payload)
+        if reset_status and job_resp_status:
+            job_uri = MANAGER_JOB_ID_URI.format(job_resp_status["Id"])
+            job_resp, job_msg = wait_for_redfish_job_complete(redfish_obj, job_uri)
+            job_status = job_resp.json_data
+            if job_status["JobState"] != "RebootCompleted" and job_msg:
+                module.fail_json(msg=job_msg)
+            elif job_status["JobState"] != "RebootCompleted" and not job_msg:
+                module.fail_json(msg=REBOOT_FAIL)
+        elif not reset_status and reset_fail:
+            module.fail_json(msg=reset_fail)
+
+        current_job_status, failed = get_job_status(redfish_obj, module, job_ids, job_wait=True)
+    if not module.params["reboot"] and preview_uri:
+        current_job_status, failed = get_job_status(redfish_obj, module, job_ids, job_wait=False)
+    if reboot_uri:
+        job_ids = simple_update(redfish_obj, reboot_uri, update_uri)
+        track, resetting, js_job_msg = wait_for_redfish_idrac_reset(module, redfish_obj,
+                                                                    module.params["reboot_timeout"])
+        if not track and resetting:
+            reboot_job_status, failed = get_job_status(redfish_obj, module, job_ids, job_wait=True)
+            current_job_status.extend(reboot_job_status)
+    return current_job_status, failed
+
+
+def main():
+    specs = {
+        "name": {"required": True, "type": "str"},
+        "reboot": {"type": "bool", "default": True},
+        "reboot_timeout": {"type": "int", "default": 3600},
+    }
+    specs.update(redfish_auth_params)
+    module = AnsibleModule(argument_spec=specs, supports_check_mode=True)
+    try:
+        with Redfish(module.params, req_session=True) as redfish_obj:
+            preview_uri, reboot_uri, update_uri = get_rollback_preview_target(redfish_obj, module)
+            job_status, failed = rollback_firmware(redfish_obj, module, preview_uri, reboot_uri, update_uri)
+            if not job_status or failed == len(job_status):
+                module.fail_json(msg=ROLLBACK_FAILED, status=job_status)
+            if module.params["reboot"]:
+                if failed > 0 and failed != len(job_status):
+                    module.fail_json(msg=COMPLETED_ERROR, status=job_status)
+                module.exit_json(msg=ROLLBACK_SUCCESS, status=job_status, changed=True)
+            else:
+                if failed > 0 and failed != len(job_status):
+                    module.fail_json(msg=SCHEDULED_ERROR, status=job_status)
+                module.exit_json(msg=ROLLBACK_SCHEDULED, status=job_status)
+    except HTTPError as err:
```

Comment:
URLError should be caught for marking the unreachable as true.

Reply:
Yes, Updated.